libs/vkd3d-shader/spirv.c

```diff
@@ -5687,6 +5716,15 @@ static uint32_t vkd3d_dxbc_compiler_get_image_type_id(struct vkd3d_dxbc_compiler
             depth, resource_type_info->arrayed, resource_type_info->ms, is_uav ? 2 : 1, format);
 }
 
+static bool vkd3d_dxbc_compiler_use_ssbo(struct vkd3d_dxbc_compiler *compiler,
+        unsigned int structure_stride, bool raw)
+{
+    /* Extract lowest bit from stride to get alignment
+     * in case the stride is not a power of two. */
+    unsigned int dword_alignment = raw ? 4 : (structure_stride & -structure_stride);
```

Comment:
Is structure stride in dwords in DXBC? It's bytes in DXIL at least.

Reply:
It is, yes.

Reply:
At least that's what it seems to be like. I tested this with a `StructuredBuffer<uint>` and we're getting 1 for `structure_stride`.

Reply:
This is why:
```c
    vkd3d_dxbc_compiler_emit_resource_declaration(compiler, reg,
            VKD3D_SHADER_RESOURCE_BUFFER, VKD3D_DATA_UINT, stride / 4, false);
```

Reply:
Ok. Weird ...